core/client/src/main/java/alluxio/client/block/stream/BlockInStream.java

```diff
@@ -77,11 +75,9 @@ public static BlockInStream createLocalBlockInStream(long blockId, long blockSiz
     try {
       BlockWorkerClient blockWorkerClient =
           closer.register(context.createBlockWorkerClient(workerNetAddress));
-      LockBlockResource lockBlockResource =
-          closer.register(blockWorkerClient.lockBlock(blockId, LockBlockOptions.defaults()));
       PacketInStream inStream = closer.register(PacketInStream
-          .createLocalPacketInStream(lockBlockResource.getResult().getBlockPath(), blockId,
-              blockSize));
+          .createLocalPacketInStream(context, blockWorkerClient.getDataServerAddress(), blockId,
+              blockWorkerClient.getSessionId(), blockSize));
```

Comment:
ack. talked in person. It might not be a significant change. but it requires to deprecate the hearbeat logic. When heartbeat is deprecated, this can be removed for both short circuit and remote IO.